ios/brave-ios/Sources/Brave/Frontend/Brave Translate/BraveTranslateTabHelper.swift

```diff
@@ -226,151 +381,232 @@ class BraveTranslateTabHelper: NSObject {
   // MARK: - Private
 
   @MainActor
-  private func executePageFunction(tab: Tab, name functionName: String) async -> String? {
-    guard let webView = tab.webView else {
-      return nil
+  private func loadTranslateScript() async throws {
+    guard let webView = tab?.webView else {
+      throw BraveTranslateError.otherError
     }
 
-    let (result, error) = await webView.evaluateSafeJavaScript(
-      functionName: "window.__firefox__.\(BraveTranslateScriptHandler.namespace).\(functionName)",
+    // Lazy loading. Load the translate script if needed.
+    _ = try? await webView.callAsyncJavaScript(
+      """
+      await window.__firefox__.\(BraveTranslateScriptHandler.namespace).loadTranslateScript();
+      """,
+      contentWorld: BraveTranslateScriptHandler.scriptSandbox
+    )
```

Comment:
This should use our safe API extension, no?

Reply:
We don't have a safe one for this call. It can't handle `promises` which `loadTranslateScript` uses, so we have to `await` in Javascript for that promise to complete, then Swift gets the result from `callAsyncJavascript`.

We'd have to write a safe extension for that, but since it's just a function call with zero parameters, it's okay.